pytest/tests/sanity/state_migration.py

```diff
@@ -74,6 +74,10 @@ def main():
 
     wait_for_blocks_or_timeout(current_node, 20, 100)
 
+    # Test deserialize state with new genesis
+    shutil.copy(os.path.join(node_root, 'test0/genesis.json'), '../near/res/testnet_genesis_config.json')
+    subprocess.check_output('cargo test --package near --lib -- config::test::test_deserialize_state --exact --nocapture', shell=True)
+    subprocess.check_output('git checkout ../near/res/testnet_genesis_config.json', shell=True)
```

Comment:
I think we should keep the new genesis config? Also we should strip records out of it, but it's probably not an issue

Reply:
Sounds good. Also I think it should be named as just "genesis_config.json" or "current_genesis_config.json" since it reflects "a genesis config that generated by `near init`" but not the value for testnet

Reply:
@ilblackdragon I think there's a good way to keep this not hacky and keep backward compatibility still indeed mean it backward compatible.
- We enforce people to update near/res/current_genesis.json to be identical to `near init` generated (except chain-id), I think genesis records is better to be kept to check they can be deserialized too.
- The test here check above
- The test here also check migrated test0_finished/output.json (moved to test0/genesis.json) without genesis records is same as near/res/current_genesis.json without genesis records.

This should be equivalent to config::test::test_deserialize_state plus requires a up to date view of current genesis in near/res/current_genesis.json

Reply:
Yeah, good point. I think we should add `near init` test separately to check if `genesis_config.json` matches output of `near init`

Reply:
Good point.

Reply:
Sounds good. I added a update_res script to update and check near/res/genesis_config.json.
And I just realized it cannot check test0_finished/output.json == near/res/genesis_config.json because the former one is generated by near init --fast